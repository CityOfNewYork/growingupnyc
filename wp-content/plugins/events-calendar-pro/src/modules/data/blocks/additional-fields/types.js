/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const SET_ADDITIONAL_FIELD_INITIAL_STATE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_INITIAL_STATE`;
export const ADD_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/ADD_ADDITIONAL_FIELD`;
export const REMOVE_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD`;

export const SET_ADDITIONAL_FIELD_LABEL = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_LABEL`;
export const SET_ADDITIONAL_FIELD_IS_PRISTINE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_IS_PRISTINE`;
export const SET_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_TYPE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_TYPE`;
export const SET_ADDITIONAL_FIELD_OPTIONS = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OPTIONS`;
export const SET_ADDITIONAL_FIELD_DIVIDER_LIST = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_LIST`;
export const SET_ADDITIONAL_FIELD_DIVIDER_END = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_END`;
export const APPEND_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/APPEND_ADDITIONAL_FIELD_VALUE`;
export const REMOVE_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_CHANGE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_CHANGE`;
export const SET_ADDITIONAL_FIELD_META_KEY = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_META_KEY`;
export const SET_ADDITIONAL_FIELD_OUTPUT = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OUTPUT`;
export const SET_ADDITIONAL_FIELD_BLUR = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_BLUR`;
