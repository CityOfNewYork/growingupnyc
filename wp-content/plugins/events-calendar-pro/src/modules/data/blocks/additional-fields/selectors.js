/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import identity from 'lodash/identity';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * Internal dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);

export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);

export const getFieldName = ( state, props ) => props.name;

export const getIds = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.allIds,
);

export const getFieldAsObjects = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.byId,
);

export const getFieldBlock = createSelector(
	[ getFieldAsObjects, getFieldName ],
	( fields, name ) => fields[ name ] || {},
);

export const getFieldDividerList = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerList,
);

export const getFieldDividerEnd = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerEnd,
);

export const getFieldOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldMetaKey = createSelector(
	[ getFieldBlock ],
	( field ) => field.metaKey,
);

export const getFieldType = createSelector(
	[ getFieldBlock ],
	( field ) => field.type,
);

export const getFieldLabel = createSelector(
	[ getFieldBlock ],
	( field ) => field.label || '',
);

export const getFieldValue = createSelector(
	[ getFieldBlock ],
	( field ) => field.value,
);

export const getTextFieldValue = createSelector(
	[ getFieldValue ],
	( value ) => value || '',
);

export const getTextAreaOutput = createSelector(
	[ getFieldOutput ],
	( output = '' ) => {
		return output.split( '\n' ).filter( identity );
	},
);

export const getFieldIsPristine = createSelector(
	[ getFieldBlock ],
	( field ) => field.isPristine,
);

export const getFieldOptions = createSelector(
	[ getFieldBlock ],
	( field ) => field.options || [],
);

export const getFieldOptionsWithLabels = createSelector(
	[ getFieldOptions ],
	( options ) => {
		return options.map( ( option ) => ( { value: option, label: option } ) );
	},
);

export const getFieldDropdownValue = createSelector(
	[ getFieldBlock ],
	( field ) => ( { value: field.value, label: field.value } ),
);

export const getFieldDropdownOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldCheckboxValue = createSelector(
	[ getTextFieldValue ],
	( value ) => uniq( value.split( '|' ) ),
);

export const getFieldCheckboxOptions = createSelector(
	[ getFieldCheckboxValue, getFieldOptionsWithLabels ],
	( values, optionsWithLabels ) => {
		return optionsWithLabels.map( ( option ) => ( {
			...option,
			isChecked: includes( values, option.value ),
		} ) );
	},
);
