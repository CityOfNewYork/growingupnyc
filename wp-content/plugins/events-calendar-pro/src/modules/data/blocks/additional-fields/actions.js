/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { SET_ADDITIONAL_FIELD_OUTPUT } from './types';

export const setInitialState = ( props ) => ( {
	type: types.SET_ADDITIONAL_FIELD_INITIAL_STATE,
	payload: props,
} );

export const setFieldChange = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_CHANGE,
	payload: {
		name,
	},
} );

export const setFieldBlur = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_BLUR,
	payload: {
		name,
	},
} );

export const setFieldBlurWithType = ( name, type ) => ( {
	type: `${ types.SET_ADDITIONAL_FIELD_BLUR }/${ type }`,
	payload: {
		name,
	},
} );

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const appendFieldValue = ( name, value ) => ( {
	type: types.APPEND_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const removeFieldValue = ( name, value ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

export const setFieldMetaKey = ( name, metaKey ) => ( {
	type: types.SET_ADDITIONAL_FIELD_META_KEY,
	payload: {
		name,
		metaKey,
	},
} );

export const setFieldOutput = ( name, output ) => ( {
	type: SET_ADDITIONAL_FIELD_OUTPUT,
	payload: {
		name,
		output,
	},
} );
