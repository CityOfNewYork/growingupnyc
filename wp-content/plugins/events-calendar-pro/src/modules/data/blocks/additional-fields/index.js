/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas };
