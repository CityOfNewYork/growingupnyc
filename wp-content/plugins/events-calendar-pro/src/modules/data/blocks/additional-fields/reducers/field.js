/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { __ } from '@wordpress/i18n';

export const DEFAULT_STATE = {
	isPristine: true,
	value: '',
	type: '',
	options: [],
	dividerList: ', ',
	dividerEnd: __( ' and ', 'tribe-events-calendar-pro' ),
	label: '',
	metaKey: '',
	output: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	const { payload = {} } = action;
	switch ( action.type ) {
		case types.SET_ADDITIONAL_FIELD_LABEL:
			return {
				...state,
				label: payload.label,
			};
		case types.SET_ADDITIONAL_FIELD_VALUE:
			return {
				...state,
				value: payload.value,
			};
		case types.SET_ADDITIONAL_FIELD_TYPE:
			return {
				...state,
				type: payload.type,
			};
		case types.SET_ADDITIONAL_FIELD_IS_PRISTINE:
			return {
				...state,
				isPristine: payload.isPristine,
			};
		case types.SET_ADDITIONAL_FIELD_OPTIONS:
			return {
				...state,
				options: payload.options,
			};
		case types.SET_ADDITIONAL_FIELD_DIVIDER_LIST:
			return {
				...state,
				dividerList: payload.dividerList,
			};
		case types.SET_ADDITIONAL_FIELD_DIVIDER_END:
			return {
				...state,
				dividerEnd: payload.dividerEnd,
			};
		case types.SET_ADDITIONAL_FIELD_META_KEY:
			return {
				...state,
				metaKey: payload.metaKey,
			}
		case types.SET_ADDITIONAL_FIELD_OUTPUT:
			return {
				...state,
				output: payload.output,
			}
		default:
			return state;
	}
}
