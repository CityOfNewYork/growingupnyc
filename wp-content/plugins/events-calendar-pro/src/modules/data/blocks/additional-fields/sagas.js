/**
 * External dependencies
 */
import { takeEvery, put, select, all } from 'redux-saga/effects';
import noop from 'lodash/noop';
import isEmpty from 'lodash/isEmpty';
import identity from 'lodash/identity';

/**
 * Internal dependencies
 */
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';
import { actions, types, selectors } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { string } from '@moderntribe/common/utils';

export function* setInitialState( props ) {
	const { payload = {} } = props;
	const { name = '', get = noop } = payload;
	const initialValues = get( 'initialValues', {} );

	yield all( [
		put( actions.addField( name ) ),
		put( actions.setFieldIsPristine( name, get( 'isPristine' ) ) ),
		put( actions.setFieldOutput( name, get( 'output' ) ) ),
		put( actions.setFieldType( name, initialValues.type ) ),
		put( actions.setFieldLabel( name, initialValues.label ) ),
		put( actions.setFieldOptions( name, initialValues.options ) ),
		put( actions.setFieldMetaKey( name, initialValues.metaKey ) ),
		put( actions.setFieldValue( name, get( 'value' ) ) ),
		put( actions.setFieldChange( name ) ),
	] );
}

export function* setPristineState( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const value = yield select( selectors.getFieldValue, payload );
	const isPristine = value === null || '' === value || isEmpty( value );
	yield put( actions.setFieldIsPristine( name, isPristine ) );
}

export function* appendFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = [ ...current, value ]
		.filter( identity )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}

export function* removeFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = current
		.filter( ( text ) => text !== value )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}

export function* onFieldBlur( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const type = yield select( selectors.getFieldType, payload );
	yield put( actions.setFieldBlurWithType( name, type ) );
}

export function* setTextFieldOutput( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const value = yield select( selectors.getFieldValue, payload );
	yield put( actions.setFieldOutput( name, value ) );
}

export function* setDropdownOutput( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const { label = '' } = yield select( selectors.getFieldDropdownValue, payload );
	yield put( actions.setFieldOutput( name, label ) );
}

export function* setCheckboxOutput( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const { values, listDivider, dividerEnd } = yield all( {
		values: select( selectors.getFieldCheckboxValue, payload ),
		listDivider: select( selectors.getFieldDividerList, payload ),
		dividerEnd: select( selectors.getFieldDividerEnd, payload ),
	} );
	yield put( actions.setFieldOutput( name, string.wordsAsList( values, listDivider, dividerEnd ) ) );
}

export default function* watchers() {
	yield takeEvery( types.SET_ADDITIONAL_FIELD_INITIAL_STATE, setInitialState );
	yield takeEvery( types.SET_ADDITIONAL_FIELD_CHANGE, setPristineState );
	yield takeEvery( types.APPEND_ADDITIONAL_FIELD_VALUE, appendFieldValue );
	yield takeEvery( types.REMOVE_ADDITIONAL_FIELD_VALUE, removeFieldValue );
	yield takeEvery( types.SET_ADDITIONAL_FIELD_BLUR, onFieldBlur );
	yield takeEvery(
		[
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.text }`,
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.radio }`,
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.url }`,
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.textarea }`,
		],
		setTextFieldOutput,
	);
	yield takeEvery(
		`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.dropdown }`,
		setDropdownOutput,
	);
	yield takeEvery(
		`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.checkbox }`,
		setCheckboxOutput,
	);
}
