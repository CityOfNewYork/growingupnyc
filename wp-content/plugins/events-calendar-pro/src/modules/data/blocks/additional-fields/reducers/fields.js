/**
 * External dependencies
 */
import { combineReducers } from 'redux';
import omit from 'lodash/omit';

/**
 * Internal dependencies
 */
import * as types from '../types';
import field from './field';

const fieldsById = ( state = {}, action ) => {
	const { payload = {} } = action;
	switch ( action.type ) {
		case types.ADD_ADDITIONAL_FIELD:
		case types.SET_ADDITIONAL_FIELD_VALUE:
		case types.SET_ADDITIONAL_FIELD_TYPE:
		case types.SET_ADDITIONAL_FIELD_OPTIONS:
		case types.SET_ADDITIONAL_FIELD_IS_PRISTINE:
		case types.SET_ADDITIONAL_FIELD_DIVIDER_LIST:
		case types.SET_ADDITIONAL_FIELD_LABEL:
		case types.SET_ADDITIONAL_FIELD_DIVIDER_END:
		case types.APPEND_ADDITIONAL_FIELD_VALUE:
		case types.REMOVE_ADDITIONAL_FIELD_VALUE:
		case types.SET_ADDITIONAL_FIELD_META_KEY:
		case types.SET_ADDITIONAL_FIELD_OUTPUT:
			return {
				...state,
				[ payload.name ]: field( state[ payload.name ], action ),
			};
		case types.REMOVE_ADDITIONAL_FIELD:
			return omit( state, [ payload.name ] );
		default:
			return state;
	}
};

const allFields = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ADDITIONAL_FIELD:
			return [ ...state, action.payload.name ];
		case types.REMOVE_ADDITIONAL_FIELD:
			return state.filter( ( name ) => name !== action.payload.name );
		default:
			return state;
	}
};

export default combineReducers( {
	byId: fieldsById,
	allIds: allFields,
} );
