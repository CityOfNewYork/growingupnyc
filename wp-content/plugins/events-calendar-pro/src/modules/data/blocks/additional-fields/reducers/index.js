/**
 * Internal dependencies
 */
import fields from './fields';

export default fields;
