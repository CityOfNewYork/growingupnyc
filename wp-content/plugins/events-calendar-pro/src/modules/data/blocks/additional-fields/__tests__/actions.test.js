/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';
import { DEFAULT_STATE } from '../reducers/field';

describe( 'Actions in Additional Fields', () => {
	const fieldName = '_ecp_custom_23';

	test( 'add field', () => {
		expect( actions.addField( fieldName ) ).toMatchSnapshot();
	} );

	test( 'remove field', () => {
		expect( actions.removeField( fieldName ) ).toMatchSnapshot();
	} );

	test( 'set field name', () => {
		expect( actions.setFieldLabel( fieldName, 'Host' ) ).toMatchSnapshot();
	} );

	test( 'set field is pristine', () => {
		expect( actions.setFieldIsPristine( fieldName, true ) ).toMatchSnapshot();
		expect( actions.setFieldIsPristine( fieldName, false ) ).toMatchSnapshot();
	} );

	test( 'set field value', () => {
		expect( actions.setFieldValue( fieldName, 'Rachel MA' ) ).toMatchSnapshot();
	} );

	test( 'set field type', () => {
		expect( actions.setFieldType( fieldName, FIELD_TYPES.checkbox ) ).toMatchSnapshot();
	} );

	test( 'set field options', () => {
		expect( actions.setFieldOptions( fieldName, [ 'Option 1', 'Option 2 ' ] ) ).toMatchSnapshot();
	} );

	test( 'set field list divider', () => {
		expect( actions.setFieldDividerList( fieldName, ', ' ) ).toMatchSnapshot();
	} );

	test( 'set field divider end', () => {
		expect( actions.setFieldDividerEnd( fieldName, ' & ' ) ).toMatchSnapshot();
	} );

	test( 'initial state', () => {
		expect( actions.setInitialState( DEFAULT_STATE ) ).toMatchSnapshot();
	} );

	test( 'on change', () => {
		expect( actions.setFieldChange( fieldName ) ).toMatchSnapshot();
	} );

	test( 'append field value', () => {
		expect( actions.appendFieldValue( fieldName, 'Chocolate' ) ).toMatchSnapshot();
	} );

	test( 'remove field value', () => {
		expect( actions.removeFieldValue( fieldName, 'Chocolate' ) ).toMatchSnapshot();
	} );

	test( 'meta key', () => {
		expect( actions.setFieldMetaKey( fieldName, '_tribe_events' ) ).toMatchSnapshot();
	} );

	test( 'set field output', () => {
		expect( actions.setFieldOutput( fieldName, 'Label on my field' ) ).toMatchSnapshot();
	} );

	test( 'set field blur', () => {
		expect( actions.setFieldBlur( fieldName ) ).toMatchSnapshot();
	} );

	test( 'set field blur with type type', () => {
		expect( actions.setFieldBlurWithType( fieldName, FIELD_TYPES.text ) ).toMatchSnapshot();
		expect( actions.setFieldBlurWithType( fieldName, FIELD_TYPES.checkbox ) ).toMatchSnapshot();
		expect( actions.setFieldBlurWithType( fieldName, FIELD_TYPES.radio ) ).toMatchSnapshot();
		expect( actions.setFieldBlurWithType( fieldName, FIELD_TYPES.dropdown ) ).toMatchSnapshot();
		expect( actions.setFieldBlurWithType( fieldName, FIELD_TYPES.textarea ) ).toMatchSnapshot();
		expect( actions.setFieldBlurWithType( fieldName, FIELD_TYPES.url) ).toMatchSnapshot();
	} );
} );
