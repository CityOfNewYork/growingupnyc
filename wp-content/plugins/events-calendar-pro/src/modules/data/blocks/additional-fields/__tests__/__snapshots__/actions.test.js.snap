// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions in Additional Fields add field 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/ADD_ADDITIONAL_FIELD",
}
`;

exports[`Actions in Additional Fields append field value 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
    "value": "Chocolate",
  },
  "type": "@@MT/EVENTS-PRO/APPEND_ADDITIONAL_FIELD_VALUE",
}
`;

exports[`Actions in Additional Fields initial state 1`] = `
Object {
  "payload": Object {
    "dividerEnd": " & ",
    "dividerList": ", ",
    "isPristine": true,
    "label": "",
    "metaKey": "",
    "options": Array [],
    "output": "",
    "type": "",
    "value": "",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_INITIAL_STATE",
}
`;

exports[`Actions in Additional Fields meta key 1`] = `
Object {
  "payload": Object {
    "metaKey": "_tribe_events",
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_META_KEY",
}
`;

exports[`Actions in Additional Fields on change 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_CHANGE",
}
`;

exports[`Actions in Additional Fields remove field 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/REMOVE_ADDITIONAL_FIELD",
}
`;

exports[`Actions in Additional Fields remove field value 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
    "value": "Chocolate",
  },
  "type": "@@MT/EVENTS-PRO/REMOVE_ADDITIONAL_FIELD_VALUE",
}
`;

exports[`Actions in Additional Fields set field blur 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR",
}
`;

exports[`Actions in Additional Fields set field blur with type type 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR/text",
}
`;

exports[`Actions in Additional Fields set field blur with type type 2`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR/checkbox",
}
`;

exports[`Actions in Additional Fields set field blur with type type 3`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR/radio",
}
`;

exports[`Actions in Additional Fields set field blur with type type 4`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR/dropdown",
}
`;

exports[`Actions in Additional Fields set field blur with type type 5`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR/textarea",
}
`;

exports[`Actions in Additional Fields set field blur with type type 6`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_BLUR/url",
}
`;

exports[`Actions in Additional Fields set field divider end 1`] = `
Object {
  "payload": Object {
    "dividerEnd": " & ",
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_DIVIDER_END",
}
`;

exports[`Actions in Additional Fields set field is pristine 1`] = `
Object {
  "payload": Object {
    "isPristine": true,
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_IS_PRISTINE",
}
`;

exports[`Actions in Additional Fields set field is pristine 2`] = `
Object {
  "payload": Object {
    "isPristine": false,
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_IS_PRISTINE",
}
`;

exports[`Actions in Additional Fields set field list divider 1`] = `
Object {
  "payload": Object {
    "dividerList": ", ",
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_DIVIDER_LIST",
}
`;

exports[`Actions in Additional Fields set field name 1`] = `
Object {
  "payload": Object {
    "label": "Host",
    "name": "_ecp_custom_23",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_LABEL",
}
`;

exports[`Actions in Additional Fields set field options 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
    "options": Array [
      "Option 1",
      "Option 2 ",
    ],
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_OPTIONS",
}
`;

exports[`Actions in Additional Fields set field output 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
    "output": "Label on my field",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_OUTPUT",
}
`;

exports[`Actions in Additional Fields set field type 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
    "type": "checkbox",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_TYPE",
}
`;

exports[`Actions in Additional Fields set field value 1`] = `
Object {
  "payload": Object {
    "name": "_ecp_custom_23",
    "value": "Rachel MA",
  },
  "type": "@@MT/EVENTS-PRO/SET_ADDITIONAL_FIELD_VALUE",
}
`;
